tests/show_techsupport/test_techsupport.py

```diff
@@ -0,0 +1,264 @@
+    duthost.command('config mirror_session add {} {} {} {} {} {} {}'
+    .format(SESSION_INFO['name'], SESSION_INFO['src_ip'], neighbor_ip,
+     SESSION_INFO['dscp'], SESSION_INFO['ttl'], SESSION_INFO['gre'], SESSION_INFO['queue']))
+
+    logger.info('Loading acl mirror rules ...')
+    load_rule_cmd = "acl-loader update full {} --session_name={}".format(acl_rule_file, SESSION_INFO['name'])
+    duthost.command('{}'.format(load_rule_cmd))
+
+    try:
+        yield
+    finally:
+        loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix='acl')
+        loganalyzer.load_common_config()
+
+        try:
+            loganalyzer.expect_regex = [LOG_EXCEPT_MIRROR_SESSION_REMOVE]
+            with loganalyzer:
+                teardown_mirroring(duthost, mirror_setup['dut_tmp_dir'])
+        except LogAnalyzerError as err:
+            raise err
+
+
+def teardown_mirroring(dut, tmp_path):
+    """
+    teardown EVERFLOW rules after test by applying empty configuration
+    :param dut: DUT host object
+    :param setup: setup information
+    :return:
+    """
+    logger.info('Removing Mirroring rules')
+    # copy rules remove configuration
+    dst = os.path.join(tmp_path, ACL_RULE_PERSISTENT_DEL_FILE)
+    dut.copy(src=os.path.join(FILES_DIR, ACL_RULE_PERSISTENT_DEL_FILE), dest=dst)
+    dut.command("acl-loader update full {}".format(dst))
+    dut.command('config mirror_session remove {}'.format(SESSION_INFO['name']))
+
+
+@pytest.fixture(scope='function', params=['acl', 'mirroring'], autouse=True)
+def config(request):
+    """
+    fixture to add configurations on setup by received parameters.
+    The parameters expected in request are the avaiable additional configurations.
+    e.g. : test_techsupport[acl]
+    """
+    return request.getfixturevalue(request.param)
+
+
+def execute_command(duthost, since):
+    """
+    Function to execute show techsupport command
+    :param duthost: DUT
+    :param since: since string enterd by user
+    """
+    stdout = duthost.command("show techsupport --since={}".format('"' + since + '"'))
+    if stdout['rc'] == SUCCESS_CODE:
+        pytest.tar_stdout = stdout['stdout']
+    return stdout['rc'] == SUCCESS_CODE
+        
+
+def test_techsupport(request, config, duthost, testbed):
+    """
+    test the "show techsupport" command in a loop
+    :param config: fixture to configure additional setups_list on dut.
+    :param duthost: DUT host
+    :param testbed: testbed
+    """
+    loop_range = request.config.getoption("--loop_num") or DEFAULT_LOOP_RANGE
+    loop_delay = request.config.getoption("--loop_delay") or DEFAULT_LOOP_DELAY
+    since = request.config.getoption("--logs_since") or str(randint(1, 23)) + " minute ago"
+
+    logger.debug("Loop_range is {} and loop_delay is {}".format(loop_range, loop_delay))
+
+    for i in range(loop_range):
+        logger.debug("Running show techsupport ... ")
+        wait_until(300, 20, execute_command, duthost, since)
```

Comment:
I don't think this line can work as you expect.
Since most of the time the "show techsupport" will hold the prompt for a little time.
And the "wait_until" will not have chance to check the elapsed time. Will not end the execution when 300 seconds is reached.